src/restclient.js

```diff
@@ -352,6 +441,44 @@ class GlpiRestClient {
         })
     }
 
+    login (userName, userPassword) {
+        return new Promise((resolve, reject) => {
+            try {
+                this.initSessionByCredentials(userName, userPassword)
+                    .then(res1 => {
+                        this.getFullSession()
+                            .then(res2 => {
+                                this.getAnItem(ITEMTYPE.User, res2.session.glpiID)
+                                    .then(res3 => {
+                                        this.getSubItems(ITEMTYPE.User, res2.session.glpiID, ITEMTYPE.UserEmail)
+                                            .then(res4 => {
+                                                resolve ({
+                                                    sessionToken: res1.session_token,
+                                                    userData: res3, 
+                                                    userEmails: res4
+                                                })
+                                            })
+                                            .catch(err4 => {
+                                                reject(err4)
+                                            })
+                                    })
+                                    .catch(err3 => {
+                                        reject(err3)
+                                    })
+                            })
+                            .catch(err2 => {
+                                reject(err2)
+                            })
+                    })
+                    .catch(err1 => {
+                        reject(err1)
+                    })
+            }
+            catch (err) {
+                reject(err)
+            }
+        })
+    }
 }
 
```

Comment:
What happened here? @Gianfranco97 
![image](https://user-images.githubusercontent.com/663460/35130395-e1c7221a-fcc0-11e7-8a8a-be76b4311970.png)

Reply:
XD

Reply:
@ajsb85 It's a chain of promises, if everything goes well I go ahead and add the next request until I have the answer I want to return to, otherwise I capture the error and return it to the user

Reply:
Here there must be an error, but while, you can not use async/await?

Reply:
@ajsb85 you're right, that would simplify the code a lot